### PR TITLE
Add MapFit test for one energy bin

### PR DESCRIPTION
**Pull request by AtreyeeS**
This PR addresses #1623 .

### CODE REVIEW

File: gammapy/cube/tests/test_fit.py [hunk 1]
```diff
@@ -99,20 +92,32 @@ def counts(sky_model, exposure, background, psf, edisp):
...
+    geom_t = geom_etrue(ebounds_true)
+
+    background_map = background(geom_r)
+    psf_map = psf(geom_t)
+    edisp_map = edisp(geom_r, geom_t)
+    exposure_map = exposure(geom_t)
+    counts_map = counts(sky_model, exposure_map, background_map, psf_map, edisp_map)
+    mask_map = mask(geom_r, sky_model)
+
     sky_model.parameters["lon_0"].value = 0.5
```
Comment (adonath) [inline gammapy/cube/tests/test_fit.py]: Could you move this code to the `sky_model` fixture?

File: gammapy/cube/tests/test_fit.py [hunk 2]
```diff
@@ -136,3 +141,53 @@ def test_cube_fit(sky_model, counts, exposure, psf, background, mask, edisp):
...
+    geom_r = geom(ebounds)
+
+    background_map = background(geom_r)
+    psf_map = psf(geom_r)
+    edisp_map = edisp(geom_r, geom_r)
+    exposure_map = exposure(geom_r)
+    counts_map = counts(sky_model, exposure_map, background_map, psf_map, edisp_map)
+    mask_map = mask(geom_r, sky_model)
+
+    sky_model.parameters["lon_0"].value = 0.5
```
Comment (adonath) [inline gammapy/cube/tests/test_fit.py]: Again move this code to the `sky_model` fixture, except for the statement to freeze the index.

File: gammapy/cube/tests/test_fit.py [hunk 3]
```diff
@@ -136,3 +141,53 @@ def test_cube_fit(sky_model, counts, exposure, psf, background, mask, edisp):
...
+        model=sky_model,
+        counts=counts_map,
+        exposure=exposure_map,
+        background=background_map,
+        mask=mask_map,
+        psf=psf_map,
+        edisp=edisp_map,
+    )
+    result = fit.run()
+
```
Comment (adonath) [inline gammapy/cube/tests/test_fit.py]: There it is sufficient to only keep `assert result.success`, the other asserts can be removed...

File: gammapy/cube/tests/test_fit.py (hunk 1 above)
Comment (AtreyeeS) [reply gammapy/cube/tests/test_fit.py]: Moving this to the `sky_model()` will also modify the values of the initial simulation.
That spoils things, right?

Comment (adonath) [reply gammapy/cube/tests/test_fit.py]: You are right of course. Alternatively you could define a second fixture `sky_model_fit`, that is used for the fitting....

Comment (AtreyeeS) [reply gammapy/cube/tests/test_fit.py]: Ok! Changes made.

**Pull request merged**